src/components/Checkout/OrderSummary.js

```diff
@@ -0,0 +1,163 @@
+import React, { useState, useEffect } from 'react';
+import {
+	Heading,
+	VStack,
+	SimpleGrid,
+	GridItem,
+	Text,
+	Button,
+	Divider,
+} from '@chakra-ui/react';
+import { colors } from '../../util/constants';
+
+import { useStripe, useElements } from '@stripe/react-stripe-js';
+
+const OrderSummary = () => {
+	const stripe = useStripe();
+	const elements = useElements();
+
+	const [message, setMessage] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		if (!stripe || !elements) {
+			// Stripe.js has not yet loaded.
+			// Make sure to disable form submission until Stripe.js has loaded.
+			return;
+		}
+
+		setIsLoading(true);
+
+		const { error } = await stripe.confirmPayment({
+			elements,
+			confirmParams: {
+				return_url: 'http://localhost:3000/checkout',
+			},
+		});
+
+		// This point will only be reached if there is an immediate error when
+		// confirming the payment. Otherwise, your customer will be redirected to
+		// your `return_url`. For some payment methods like iDEAL, your customer will
+		// be redirected to an intermediate site first to authorize the payment, then
+		// redirected to the `return_url`.
+		if (error.type === 'card_error' || error.type === 'validation_error') {
+			setMessage(error.message);
+		} else {
+			setMessage('An unexpected error occured.');
+		}
+
+		setIsLoading(false);
+	};
+
+	useEffect(() => {
+		if (!stripe) {
+			return;
+		}
+
+		const clientSecret = new URLSearchParams(window.location.search).get(
+			'payment_intent_client_secret'
+		);
+
+		if (!clientSecret) {
+			return;
+		}
+
+		stripe.retrievePaymentIntent(clientSecret).then(({ paymentIntent }) => {
+			switch (paymentIntent.status) {
+				case 'succeeded':
+					setMessage('Payment succeeded!');
+					break;
+				case 'processing':
+					setMessage('Your payment is processing.');
+					break;
+				case 'requires_payment_method':
+					setMessage('Your payment was not successful, please try again.');
+					break;
+				default:
+					setMessage('Something went wrong.');
+					break;
+			}
+		});
+	}, [stripe]);
+
+	return (
+		<VStack
+			w='full'
+			p={10}
+			spacing={10}
+			alignItems='flex-start'
+			style={{
+				border: '2px solid #E2E8F0',
+				borderRadius: '25px',
+			}}
+		>
+			<Heading size='xl'>Order Summary</Heading>
+			<SimpleGrid columns={2} rowGap={6} columnGap={2} w='full'>
+				<GridItem colSpan={1}>
+					<Text>{`Items(00):`}</Text>
```

Comment:
You might be hard-coding things here. Let's utilize props for these.

Reply:
All the info for the hard coding part will be coming from cart object which is accessible through context api. Since we haven't implemented that,  i decided to hard code these data such as tax, item, price etc. It wont be coming from prop